aten/src/ATen/native/Copy.cpp

```diff
@@ -38,8 +38,8 @@ namespace at {
 namespace native {
 
 Tensor& _s_copy__cpu(Tensor& self, const Tensor& src, bool non_blocking) {
-  if (src.is_cuda()) {
-    _s_copy_from(src, self, non_blocking);
+  if (src.is_cuda() || src.is_xla()) {
+    self = _s_copy_from(src, self, non_blocking);
```

Comment:
do you actually need to save self here?  I think this is basically a no-op (it should swap the TensorImpl of the parameter self with the TensorImpl of the Tensor returned from `_s_copy_from`, which should be the same.

Reply:
How do we swap if we have a const&?
Is there a "swap" API which is const?
Our APIs build at::Tensor s.
Since _s_copy_from() returns the rewritten tensor (AFAICT from the CUDA implementation - which does some tricks by itself to get around the const&), so assigning self allows for clean API w/out breaking existing semantics.